Assets/Scripts/UI/TimerCountdown.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using Events;
+using TMPro;
+using UnityEngine;
+using EventType = Events.EventType;
+
+namespace UI
+{
+    public class TimerCountdown : MonoBehaviour
+    {
+        [SerializeField] private TextMeshProUGUI text;
+        private float duration;
+        private bool startedTimer;
+
+        private void FixedUpdate()
+        {
+            DisplayTime();
+        }
+
+        //Updates and shows the time remaining
+        private void DisplayTime()
+        {
+            duration -= Time.deltaTime;
+
+            var minutes = Mathf.FloorToInt(duration / 60);
+            var seconds = Mathf.FloorToInt(duration % 60);
+            text.text = minutes + ":" + seconds;
+
+            CheckIfTimeExpired();
+        }
+
+        private void CheckIfTimeExpired()
+        {
+            if (duration<=0&&startedTimer)
+                EventManager.currentManager.AddEvent(new TimerExpired());
```

Comment:
This would constantly fire off the event after the timer expires, is this intended?

Reply:
Do not worry about this, this has to be reworked anyway because i did the functionality wrong